health-services/project/src/main/java/org/egov/project/repository/rowmapper/UserActionRowMapper.java

```diff
@@ -0,0 +1,93 @@
+     * @param rowNum    The number of the current row.
+     * @return A UserAction object populated with data from the current row of the ResultSet.
+     * @throws SQLException If there is an issue accessing the ResultSet data or processing JSON.
+     */
+    @Override
+    public UserAction mapRow(ResultSet resultSet, int rowNum) throws SQLException {
+        try {
+            // Creating AuditDetails object with information from the ResultSet
+            AuditDetails auditDetails = AuditDetails.builder()
+                    .createdBy(resultSet.getString("createdBy"))
+                    .createdTime(resultSet.getLong("createdTime"))
+                    .lastModifiedBy(resultSet.getString("lastModifiedBy"))
+                    .lastModifiedTime(resultSet.getLong("lastModifiedTime"))
+                    .build();
+
+            // Creating client-specific AuditDetails object with information from the ResultSet
+            AuditDetails clientAuditDetails = AuditDetails.builder()
+                    .createdTime(resultSet.getLong("clientCreatedTime"))
+                    .createdBy(resultSet.getString("clientCreatedBy"))
+                    .lastModifiedTime(resultSet.getLong("clientLastModifiedTime"))
+                    .lastModifiedBy(resultSet.getString("clientLastModifiedBy"))
+                    .build();
+
+            // Building the UserAction object with data from the ResultSet
+            UserAction userAction = UserAction.builder()
+                    .id(resultSet.getString("id"))
+                    .tenantId(resultSet.getString("tenantId"))
+                    .clientReferenceId(resultSet.getString("clientReferenceId"))
+                    .projectId(resultSet.getString("projectId"))
+                    .latitude(resultSet.getDouble("latitude"))
+                    .longitude(resultSet.getDouble("longitude"))
+                    .locationAccuracy(resultSet.getDouble("locationAccuracy"))
+                    .boundaryCode(resultSet.getString("boundaryCode"))
+                    .action(TaskAction.fromValue(resultSet.getString("action")))
+                    .status(resultSet.getString("status"))
+                    .beneficiaryTag(resultSet.getString("beneficiaryTag"))
+                    .resourceTag(resultSet.getString("resourceTag"))
+                    .rowVersion(resultSet.getInt("rowVersion"))
+                    .auditDetails(auditDetails)
+                    .clientAuditDetails(clientAuditDetails)
+                    .additionalFields(resultSet.getString("additionalDetails") == null ? null : objectMapper
+                            .readValue(resultSet.getString("additionalDetails"), AdditionalFields.class))
+                    .build();
+
+            return userAction;
+        } catch (JsonProcessingException e) {
+            // Throwing a SQLException if there's an error processing JSON
+            log.error("Error processing JSON for UserAction mapping. Row number: " + rowNum, e);
```

Comment:
can use primary key instead of row num, that should be a better error msg

Reply:
done